pkg/ir/spec.go

```diff
@@ -16,39 +16,50 @@ import (
 )
 
 type (
-	ID          = string
-	NamedEntity string
+	ID           = string
+	NamedEntity  string
+	ResourceType string
 
 	Spec struct {
 		// Required connections
-		Connections []Connection
+		Connections []*Connection
 
-		Defs Definitions
+		Defs *Definitions
+
+		*BlockedResources
 	}
 
 	Connection struct {
 		// Egress resource
-		Src Resource
+		Src *LocalRemotePair
 
 		// Ingress resource
-		Dst Resource
+		Dst *LocalRemotePair
 
 		// Allowed protocols
-		TrackedProtocols []TrackedProtocol
+		TrackedProtocols []*TrackedProtocol
 
 		// Provenance information
 		Origin fmt.Stringer
 	}
 
-	Resource struct {
+	// LocalRemotePair holds a local resource and the remote CIDRs it should be connected to
+	LocalRemotePair struct {
 		// Symbolic name of resource, if available
-		Name string
+		Name *string
+
+		LocalCidrs []*NamedAddrs
 
-		// list of CIDR / Ip addresses.
-		IPAddrs []*netset.IPBlock
+		// Cidr list
+		RemoteCidrs []*NamedAddrs
+
+		// LocalType of resource
+		LocalType ResourceType
+	}
 
-		// Type of resource
-		Type ResourceType
+	NamedAddrs struct {
+		IPAddrs *netset.IPBlock
+		Name    *string
```

Comment:
why `Name` should be a pointer to a string?
Seems simpler if it's not

Reply:
What should we save a string instead of a ptr?